### ISSUE TITLE
Add Views to Toolbar added to CollapsingToolbarLayout

### ISSUE THREAD

**Issue by DHosseiny** [material-components/material-components-android#1867]
**Description:** The problem is when I add a view for Exmaple an ImageButton to toolbar that is direct child of CollapsingToolbarLayout that ImageButton's layout_gravity is not true in Rtl languages. When I set layout_gravity to "start" for exmaple, this ImageButton's is sticked to left of toolbar in English language(as a Ltr language) but in persian language(as a Rtl language) it is sticked to left of toolbar again(as this means "end" not "start").

**Expected behavior:** layout_gravity="start" of direct child of toolbar in CollapsingToolbarLayout  should stick to right of toolbar and vice versa.

**Source code:** 

**Android API version:** Any but tested in 29

**Material Library version:** 1.2.1 latest stable right now

**Device:** Nexus 4 emulator

**Comment by DHosseiny** [comment]
I tested toolbar's `titleMarginStart`  attr and it has the same bug(start converts to left in Rtl languages).

**Comment by matpag** [comment]
Can you try using `MaterialToolbar` and inflating a menu instead of adding ImageViews as Toolbar's children?
For example, check the layout [here](https://github.com/material-components/material-components-android/blob/b1ef02a16b78352702722550732e06758fd068cb/catalog/java/io/material/catalog/topappbar/res/layout/cat_topappbar_toolbar_fragment.xml) and check the inflation code [here](https://github.com/material-components/material-components-android/blob/b1ef02a16b78352702722550732e06758fd068cb/catalog/java/io/material/catalog/topappbar/TopAppBarToolbarDemoFragment.java#L55-L65)

**Comment by DHosseiny** [comment]
@matpag I fixed my own issue but I am looking for a fix on this.

**Comment by matpag** [comment]
If you are referring to Toolbar icons, I don't think there is something to fix here. 
Adding ImageViews directly inside Toolbar is not supported and not suggested anywhere I can find online, pretty every tutorial suggest you to add icons via menu inflation.

If you are talking about `titleMarginStart` I did not verified it yet.

**Comment by DHosseiny** [comment]
Yes, `titleMarginStart ` being bugged in RTL.(I think if this get fixed the main problem will be fixed too)
